### DISCUSSION TITLE
How to filter on JSON column

### DISCUSSION THREAD

**Discussion by n4rk0o** [piccolo-orm/piccolo#910] [Q&A] [answered]
Hello,

I started using Piccolo ORM and everything is working like a charm except for one thing. I'm trying to filter objects by using the Where clause but I'm unable to find a way to filter on a JSON column and more precisely on a specific key (like {'name': 'Paris'})

Is is possible and if yes how can I do that?

Thanks a lot and keep continue you are doing a geat job .

**Comment by dantownsend** [comment, marked as answer]
@n4rk0o Thanks for the kind words.

For filtering `JSONB` columns, you can try the [arrow](https://piccolo-orm.readthedocs.io/en/latest/piccolo/schema/column_types.html#arrow) function.

For a table like this:

```python
class RecordingStudio(Table):
    facilities = JSONB()
```

You can do:

```python
await RecordingStudio.select().where(RecordingStudio.facilities.arrow('foo') != None)
```

This doesn't work on `JSON` columns at the moment. - for `JSON` columns you can use [WhereRaw](https://piccolo-orm.readthedocs.io/en/latest/piccolo/query_clauses/where.html#whereraw). 

```python
from piccolo.query import WhereRaw

await RecordingStudio.select().where(WhereRaw("facilities->>'some key' IS NOT NULL"))
```

**Reply by n4rk0o** [reply]
Hello @dantownsend 

Thanks for the quick update I appreciate.

**Comment by n4rk0o** [comment]
Hello @dantownsend could you explain me how to filter on a sub key on a JSONB object?

Example:
Column name is "animal"
Value: {"dog": {"race":"labrador"}}

I need to filter on race key.

I tried with arrows but it's failing.

Thanks for your help 🙂

**Reply by n4rk0o** [reply]
I finally found how to do:

```python
from piccolo.query import WhereRaw

await RecordingStudio.select().where(WhereRaw("facilities->'some key'->'sub key'->>'final_key' IS NOT NULL"))
```
